test/test_distributions.py

```diff
@@ -49,8 +50,10 @@
     sum_rightmost,
     vec_to_tril_matrix,
 )
+from numpyro.infer import MCMC, NUTS, Predictive
```

Comment:
It is better to avoid adding inference tests in this file. Maybe simple tests like are enough?
```
def test_jitted_lazy_property():
    d = dist.TruncatedNormal(...)
    jax.vmap(d.log_prob)(jnp.ones(3))
```

Reply:
Somewhat related: since the key change is to the `distributions.util` module, I might suggest putting this test in 
`test_distributions_util.py` rather than `test_distributions.py`. Do you agree, @fehiepsi?

Reply:
I like this idea, but at least for me the specific suggested test passes even with the existing `lazy_property` implementation (I have confirmed that the MRE from #1836 does indeed error).

Reply:
maybe calling vmap 2 times? it is surprised to me that jax does not catch this

Reply:
re distributiomz_util: that works :)

Reply:
Have proposed a pair of tests in https://github.com/sbidari/numpyro/pull/1 and confirmed they fail with the latest `numpyro` on PyPI but pass with @sbidari's patch.

Reply:
Thanks @dylanhmorris for rewriting the tests! Ready for review @fehiepsi